subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java

```diff
@@ -224,12 +225,15 @@ public static void configureDocumentationVariantWithArtifact(String variantName,
         capabilities.forEach(variant.getOutgoing()::capability);
 
         if (!tasks.getNames().contains(jarTaskName)) {
-            tasks.register(jarTaskName, Jar.class, jar -> {
+            TaskProvider<Jar> jarTask = tasks.register(jarTaskName, Jar.class, jar -> {
                 jar.setDescription("Assembles a jar archive containing the " + (featureName == null ? "main " + docsType + "." : (docsType + " of the '" + featureName + "' feature.")));
                 jar.setGroup(BasePlugin.BUILD_GROUP);
                 jar.from(artifactSource);
                 jar.getArchiveClassifier().set(TextUtil.camelToKebabCase(featureName == null ? docsType : (featureName + "-" + docsType)));
             });
+            if (tasks.getNames().contains(LifecycleBasePlugin.ASSEMBLE_TASK_NAME)) {
```

Comment:
I have been really conservative there. If you believe we can assume `assemble` is always there, the `if` can be removed.

Reply:
I would still keep that here